# **Data Extraction**

## Objectives

* Load the car price dataset and take a first look at the data to understand what information it contains.

## Inputs

* Car price dataset
('CarPrice_Assignment.csv') from the 'Data' folder

## Outputs

* The car price dataset is saved in the  'Data' folder and ready to use for the next stage and analysis.





---

# Change working directory

* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [1]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Users\\liene\\OneDrive\\Documents\\Car Price Analysis\\Car-Prices-Analysis-Project-1\\jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [2]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\liene\\OneDrive\\Documents\\Car Price Analysis\\Car-Prices-Analysis-Project-1'

# Load the Dataset

The dataset is loaded from Data folder, ready to explore and analyse it.

In [6]:
import pandas as pd

#load the car price dataset
df = pd.read_csv("Data/CarPrice_Assignment.csv")

---

# First Look at the Dataset

Explore the dataset and its content.

In [7]:
# Display the first 5 rows of the dataset
df.head()

,car_ID,symboling,CarName,fueltype,aspiration,doornumber,carbody,drivewheel,enginelocation,wheelbase,...,enginesize,fuelsystem,boreratio,stroke,compressionratio,horsepower,peakrpm,citympg,highwaympg,price
0,1,3,alfa-romero giulia,gas,std,two,convertible,rwd,front,88.6,...,130,mpfi,3.47,2.68,9.0,111,5000,21,27,13495.0
1,2,3,alfa-romero stelvio,gas,std,two,convertible,rwd,front,88.6,...,130,mpfi,3.47,2.68,9.0,111,5000,21,27,16500.0
2,3,1,alfa-romero Quadrifoglio,gas,std,two,hatchback,rwd,front,94.5,...,152,mpfi,2.68,3.47,9.0,154,5000,19,26,16500.0
3,4,2,audi 100 ls,gas,std,four,sedan,fwd,front,99.8,...,109,mpfi,3.19,3.40,10.0,102,5500,24,30,13950.0
4,5,2,audi 100ls,gas,std,four,sedan,4wd,front,99.4,...,136,mpfi,3.19,3.40,8.0,115,5500,18,22,17450.0


### Dataset Size

Check how many rows and collumns are in the dataset.

In [8]:
# Check the size of the dataset
df.shape

(205, 26)

---

### Column names

Check the collumn names in the dataset

In [9]:
# Check the column names in the dataset
df.columns

Index(['car_ID', 'symboling', 'CarName', 'fueltype', 'aspiration',
       'doornumber', 'carbody', 'drivewheel', 'enginelocation', 'wheelbase',
       'carlength', 'carwidth', 'carheight', 'curbweight', 'enginetype',
       'cylindernumber', 'enginesize', 'fuelsystem', 'boreratio', 'stroke',
       'compressionratio', 'horsepower', 'peakrpm', 'citympg', 'highwaympg',
       'price'],
      dtype='object')

### Data types
Check the data type of each column in the dataset.

In [12]:
# Check the data types of each column
df.dtypes

car_ID                int64
symboling             int64
CarName              object
fueltype             object
aspiration           object
doornumber           object
carbody              object
drivewheel           object
enginelocation       object
wheelbase           float64
carlength           float64
carwidth            float64
carheight           float64
curbweight            int64
enginetype           object
cylindernumber       object
enginesize            int64
fuelsystem           object
boreratio           float64
stroke              float64
compressionratio    float64
horsepower            int64
peakrpm               int64
citympg               int64
highwaympg            int64
price               float64
dtype: object

# Conclusions and Next steps.

* The dataset contains 205 rows and 26 columns with a mixture  of  different data. 
* The enext step is to clean the dataset and check for any missing values and duplicates. 